### PR TITLE
feat: 🎸 Added descriptive message when using field-level auth

### PR DESCRIPTION
**Pull request by ncarvajalc**
Updated the error message when using field-level authorization rules with required fields and redirect to docs for more information

✅ Closes: #971

#### Description of changes
Changed message described in #971 to:
image
 

#### Issue #, if available
#971

#### Description of how you validated changes

Created demo project with the following schema:
```graphql
# This "input" configures a global authorization rule to enable public access to
# all models in this schema. Learn more about authorization rules here: https://docs.amplify.aws/cli/graphql/authorization-rules
input AMPLIFY {
  globalAuthRule: AuthRule = { allow: public }
} # FOR TESTING ONLY!
# Put schemas below this line
type ErrorModel @model {
  id: ID!
  requiredField: String!
  fieldWithAuthRules: String
    @auth(
      rules: [
        { allow: private, operations: [read] }
        { allow: groups, groups: ["Group"] }
      ]
    )
}
```

And `amplify-dev` with the changes.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts [hunk 1]
```diff
@@ -107,6 +107,34 @@ import {
...
+ * if we have a rule like cognito private we can remove all other related roles from the field since it has top level
+ * access by the provider
+ */
+const getReadRolesForField = (acm: AccessControlMatrix, readRoles: Array<string>, fieldName: string): Array<string> => {
+  const hasCognitoPrivateRole = readRoles.some(r => r === 'userPools:private')
+    && acm.isAllowed('userPools:private', fieldName, 'get')
+    && acm.isAllowed('userPools:private', fieldName, 'list')
+    && acm.isAllowed('userPools:private', fieldName, 'sync')
+    && acm.isAllowed('userPools:private', fieldName, 'search')
+    && acm.isAllowed('userPools:private', fieldName, 'listen');
```
Comment (alharris-at) [inline packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts:121]: I'd rewrite this in a slightly more functional way. You could convert this to a small function like 

```typescript
const hasFullReadAccessForRole = (acm: AccessControlMatrix, readRoles: Array<string>, fieldName: string, role: string): boolean => {
  return readRoles.some(r => r === role) && READ_PERMISSIONS.every(p => acm.isAllowed(role, fieldName, p);
};
```

Comment (alharris-at) [reply packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts:121]: Also, is the readRoles.som check required if we're then verifying the acm against all read access types?

Comment (alharris-at) [reply packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts:121]: Ahh nvm, I just saw you moved this. ignore these comments.

File: packages/amplify-graphql-auth-transformer/package.json [hunk 2]
```diff
@@ -32,6 +32,7 @@
     "@aws-amplify/graphql-relational-transformer": "0.12.4",
     "@aws-amplify/graphql-transformer-core": "0.17.15",
     "@aws-amplify/graphql-transformer-interfaces": "1.14.9",
+    "@aws-cdk/aws-iam": "^1.181.0",
```
Comment (alharris-at) [inline packages/amplify-graphql-auth-transformer/package.json:35]: Why the new dependency?

File: packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts [hunk 3]
```diff
@@ -190,7 +218,8 @@ export class AuthTransformer extends TransformerAuthBase implements TransformerA
     this.addTypeToResourceReferences(def.name.value, this.rules);
     // turn rules into roles and add into acm and roleMap
     this.convertRulesToRoles(acm, this.rules, isJoinType, undefined, undefined, context);
-    this.modelDirectiveConfig.set(typeName, getModelConfig(modelDirective, typeName, context.featureFlags, context.isProjectUsingDataStore()));
+    this.modelDirectiveConfig.set(typeName,
+      getModelConfig(modelDirective, typeName, context.featureFlags, context.isProjectUsingDataStore()));
```
Comment (alharris-at) [inline packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts:222]: Not a blocker, but just a n.p. There are a few formatting/newline changes in this CR that distract from the fact that you're basically just changing the error string.

**Review by alharris-at — APPROVED** [review]

File: packages/amplify-graphql-auth-transformer/package.json (hunk 2 above)
Comment (ncarvajalc) [reply packages/amplify-graphql-auth-transformer/package.json:35]: It wouldn't let me push without the dependency addition

File: packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts (hunk 3 above)
Comment (ncarvajalc) [reply packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts:222]: Yeap. The thing was I would get a warning every time I tried to `yarn commit`. It pointed out those es-lint errors and `yarn lint-fix` wouldn't fix them

**Review by phani-srikar — APPROVED** [review]

File: packages/amplify-graphql-auth-transformer/package.json (hunk 2 above)
Comment (phani-srikar) [reply packages/amplify-graphql-auth-transformer/package.json:35]: What's the error that you're seeing and when does it occur?

**Review by phani-srikar — CHANGES_REQUESTED** [review]
LGTM! just want to make sure we're not adding the extra dependency that @alharris-at pointed out.

File: packages/amplify-graphql-auth-transformer/package.json (hunk 2 above)
Comment (alharris-at) [reply packages/amplify-graphql-auth-transformer/package.json:35]: @ncarvajalc could you flip this to `~1.172.0`.

**Review by phani-srikar — APPROVED** [review]

**Review by alharris-at — APPROVED** [review]

**Pull request merged**